### ISSUE TITLE
Error in BIOMOD_xxx - [error in bm_ModelingOptions]

### ISSUE THREAD

**Issue by zali37** [biomodhub/biomod2#686]
_Please make sure to close the issue once you consider it as solved_
_Please use screenshots only when you cannot copy-paste the object, e.g. for figures or maps_

**Error and context**
Hello, biomod team. I am continuing to modify my original code to include pseudo-absences instead of true absences and, in turn, add two more algorithms. However, my old bm_ModelingOptions script is not functioning. I am getting an error that appears to be a writing error, but I do not know how to fix it. 
**Code used to get the error**
bibronii_format <- BIOMOD_FormatingData(
  resp.name = "bibronii",
  resp.var = bibronii[,"presence"],
  resp.xy = bibronii[, c("lon", "lat")],
  expl.var = envs,
  PA.nb.rep = 5,
  PA.nb.absences = 200,
  PA.strategy = "disk",
  PA.dist.min = -1,
  PA.dist.max = -3,
  filter.raster = T,
  seed.val = 4
)
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-= BIOMOD.formated.data -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

dir.name =  .

sp.name =  bibronii

	 131 presences,  0 true absences and  981 undefined points in dataset

	 7 explanatory variables

   clip_bio3       clip_bio6        clip_bio8        clip_bio9        clip_bio14    
 Min.   :36.50   Min.   :-9.632   Min.   :-4.080   Min.   :-5.216   Min.   :  3.00  
 1st Qu.:46.15   1st Qu.:-3.418   1st Qu.: 2.997   1st Qu.: 6.950   1st Qu.:  9.00  
 Median :46.63   Median :-2.040   Median : 4.896   Median :10.755   Median : 11.00  
 Mean   :46.42   Mean   :-2.076   Mean   : 4.977   Mean   :10.163   Mean   : 26.21  
 3rd Qu.:47.20   3rd Qu.:-0.296   3rd Qu.: 6.667   3rd Qu.:13.634   3rd Qu.: 21.00  
 Max.   :53.43   Max.   : 2.940   Max.   :14.150   Max.   :18.197   Max.   :179.00  
   clip_bio15     pedregal_continuoFINAL
 Min.   : 8.315   Min.   :0.0000000     
 1st Qu.:21.079   1st Qu.:0.0000000     
 Median :26.306   Median :0.0002987     
 Mean   :27.205   Mean   :0.4162051     
 3rd Qu.:32.323   3rd Qu.:0.3285952     
 Max.   :59.618   Max.   :4.6952438     

 5 Pseudo Absences dataset available ( PA1, PA2, PA3, PA4, PA5 ) with  
200 (PA1, PA2, PA3, PA4, PA5) pseudo absences

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
cv.k <- bm_CrossValidation(bm.format = bibronii_format,
                           strategy = "block")

opt.d <- bm_ModelingOptions(data.type = 'binary',
                            models = c("RF","ANN","GLM","SRE","MAXENT",
                                       "GAM.gam.gam"),
                            strategy = "default",
                            bm.format = bibronii_format,
                            calib.lines = cv.k)

user.rf <- list("_allData_allRun"=list(
  mtry=2,
  type = "classification",
  ntre=500,
  replace = T,
  nodesize=10))

tuned.gam <- bm_Tuning(model = 'GAM',
                       tuning.fun = 'gam',
                       do.stepAIC = TRUE,
                       bm.options = opt.d@options$GAM.binary.gam.gam,
                       bm.format = bibronii_format)

tuned.ann <- bm_Tuning(model = "ANN",
                       tuning.fun = "avNNet",
                       bm.options = opt.d@options$ANN.binary.nnet.nnet,
                       bm.format = bibronii_format
                         )

opt.sre <- list("_allData_allRun"=list(quant=0.1))

occs <- read.csv(file = "soloPresencias_bibroniiMAXENT.csv",header = T,sep = ",",dec = ".")

e.maxent <- ENMevaluate(occs,
                        envs,
                        n.bg = 200,
                        tune.args = list(fc=c("L","Q","LQ","LQH"),rm=1:5),
                        partitions = "randomkfold",
                        algorithm = "maxent.jar",
                        partition.settings = list(kfolds=3),
                        other.settings = list(
                          path="C:/DATOS/ENM/diplolaemus",
                          removeduplicates=T,
                          jacknife=T,
                          responsecurves=T,
                        taxon.name= "bibronii",
                        rmm=T)
                        )

form.GLM <- bm_MakeFormula(resp.name = "bibronii",
                           expl.var = envs_GLM,
                           type = 'quadratic',
                           interaction.level = 2)
user.GLM <- list('_allData_allRUN' = list(formula = form.GLM))

bibronii_opt <- bm_ModelingOptions(
  data.type = "binary",
  models=c("SRE","RF", "GAM.gam.gam","ANN","GLM","MAXENT"),
  bm.format = bibronii_format,
  strategy = "user.defined",
  user.base = "bigboss",
  user.val = list(SRE.binary.biomod2.bm_SRE = opt.sre,
                  RF.binary.randomForest.randomForest = user.rf,
                  GAM.binary.gam.gam = tuned.gam,
                  ANN.binary.nnet.nnet = tuned.ann,
                  GLM.binary.stats.glm = user.GLM,
                  MAXENT.binary.MAXENT.MAXENT = e.maxent)
)

```
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-= Build Modeling Options -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

	>  SRE options (datatype: binary , package: biomod2 , function: bm_SRE )...
	>  RF options (datatype: binary , package: randomForest , function: randomForest )...
	>  GAM options (datatype: binary , package: gam , function: gam )...
	>  ANN options (datatype: binary , package: nnet , function: nnet )...
	>  GLM options (datatype: binary , package: stats , function: glm )...
	>  MAXENT options (datatype: binary , package: MAXENT , function: MAXENT )...
Error in { : task 5 failed - "task 1 failed - "
names(user.val) must be '_allData_allRun', '_PA1_allRun', '_PA2_allRun', '_PA3_allRun', '_PA4_allRun', '_PA5_allRun' or 'for_all_datasets""
In addition: Warning messages:
1: In .BIOMOD.options.dataset.check.args(strategy = strategy, user.val = user.val,  :
  Options will be changed only for a subset of datasets (_allData_allRun) and not the others (_PA1_allRun, _PA2_allRun, _PA3_allRun, _PA4_allRun, _PA5_allRun). 
Please update 'user.val' argument if this is not wanted.
2: In .BIOMOD.options.dataset.check.args(strategy = strategy, user.val = user.val,  :
  Options will be changed only for a subset of datasets (_allData_allRun) and not the others (_PA1_allRun, _PA2_allRun, _PA3_allRun, _PA4_allRun, _PA5_allRun). 
Please update 'user.val' argument if this is not wanted.
3: In .BIOMOD.options.dataset.check.args(strategy = strategy, user.val = user.val,  :
  Options will be changed only for a subset of datasets (_PA1_allRun, _PA2_allRun, _PA3_allRun, _PA4_allRun, _PA5_allRun) and not the others (_allData_allRun). 
Please update 'user.val' argument if this is not wanted.
4: In .BIOMOD.options.dataset.check.args(strategy = strategy, user.val = user.val,  :
  Options will be changed only for a subset of datasets (_PA1_allRun, _PA2_allRun, _PA3_allRun, _PA4_allRun, _PA5_allRun) and not the others (_allData_allRun). 
Please update 'user.val' argument if this is not wanted.

```

**Environment Information**
R version 4.5.2 (2025-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26200)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Buenos_Aires
tzcode source: internal

attached base packages:
[1] parallel  splines   stats     graphics  grDevices utils     datasets  methods  
[9] base     

other attached packages:
 [1] caret_7.0-1          ENMeval_2.0.5.2      sampling_2.11        dplyr_1.1.4         
 [5] ggplot2_4.0.1        snowfall_1.84-6.3    snow_0.4-4           biomod2_4.2-6-2     
 [9] xgboost_3.1.2.1      randomForest_4.7-1.2 maxnet_0.1.4         earth_5.3.4         
[13] plotmo_3.6.4         plotrix_3.8-13       Formula_1.2-5        gbm_2.2.2           
[17] lattice_0.22-7       doParallel_1.0.17    iterators_1.0.14     ggtext_0.1.2        
[21] tidyterra_0.7.2      mgcv_1.9-3           nlme_3.1-168         gam_1.22-6          
[25] foreach_1.5.2        raster_3.6-32        sp_2.2-0             mda_0.5-5           
[29] class_7.3-23         rpart_4.1.24         nnet_7.3-20          terra_1.8-86        

loaded via a namespace (and not attached):
 [1] DBI_1.2.3                pROC_1.19.0.1            rlang_1.1.6             
 [4] magrittr_2.0.4           predicts_0.1-19          e1071_1.7-16            
 [7] compiler_4.5.2           vctrs_0.6.5              reshape2_1.4.5          
[10] stringr_1.6.0            pkgconfig_2.0.3          prodlim_2025.04.28      
[13] rangeModelMetadata_0.1.5 purrr_1.2.0              jsonlite_2.0.0          
[16] PresenceAbsence_1.1.11   recipes_1.3.1            reshape_0.8.10          
[19] R6_2.6.1                 stringi_1.8.7            RColorBrewer_1.1-3      
[22] RPostgreSQL_0.7-8        parallelly_1.46.0        lubridate_1.9.4         
[25] Rcpp_1.1.0               future.apply_1.20.1      timechange_0.3.0        
[28] Matrix_1.7-4             tidyselect_1.2.1         rstudioapi_0.18.0       
[31] abind_1.4-8              timeDate_4051.111        codetools_0.2-20        
[34] listenv_0.10.0           tibble_3.3.0             plyr_1.8.9              
[37] withr_3.0.2              S7_0.2.1                 BIEN_1.2.7              
[40] future_1.68.0            survival_3.8-3           sf_1.0-23               
[43] rJava_1.0-14             units_1.0-0              proxy_0.4-27            
[46] xml2_1.5.1               lpSolve_5.6.23           pillar_1.11.1           
[49] KernSmooth_2.23-26       stats4_4.5.2             generics_0.1.4          
[52] scales_1.4.0             globals_0.18.0           glue_1.8.0              
[55] tools_4.5.2              ecospat_4.1.3            data.table_1.17.8       
[58] ModelMetrics_1.2.2.2     gower_1.0.2              grid_4.5.2              
[61] tidyr_1.3.1              ape_5.8-1                ipred_0.9-15            
[64] cli_3.6.5                lava_1.8.2               gtable_0.3.6            
[67] digest_0.6.39            classInt_0.4-11          farver_2.1.2            
[70] lifecycle_1.0.4          hardhat_1.4.2            gridtext_0.1.5          
[73] MASS_7.3-65             
```
Any suggestions are welcome.
Best regards, 
Natalia

**Comment by MayaGueguen** [comment]
Hello Natalia,

Ah, my bad, I forgot to check that.
It is indeed not working currently with `MAXENT.algorithm = 'maxent.jar'` but `MAXENT.algorithm = 'maxnet'` should be working fine.

Maya
